daemon/oci_windows.go

```diff
@@ -25,11 +25,31 @@ func (daemon *Daemon) createSpec(c *container.Container) (*specs.Spec, error) {
 	// In base spec
 	s.Hostname = c.FullHostname()
 
+	if err := daemon.setupSecretDir(c); err != nil {
+		return nil, err
+	}
+
 	// In s.Mounts
 	mounts, err := daemon.setupMounts(c)
 	if err != nil {
 		return nil, err
 	}
+
+	if !c.HasBeenStartedBefore && len(c.SecretReferences) > 0 {
+		if err := daemon.Mount(c); err != nil {
```

Comment:
Yes, for Windows Server Containers, you are guaranteed that the container FS is mounted here.  In start.go:

```golang
	if err := daemon.conditionalMountOnStart(container); err != nil {
		return err
	}

	if err := daemon.initializeNetworking(container); err != nil {
		return err
	}

	spec, err := daemon.createSpec(container)
	if err != nil {
		return err
	}
```

Where `conditionalMountOnStart` always mounts Windows Server Containers, but not Hyper-V containers:

```golang
	// We do not mount if a Hyper-V container
	if !daemon.runAsHyperVContainer(container.HostConfig) {
		return daemon.Mount(container)
	}
```